templates/settings.py.erb

```diff
@@ -40,6 +40,15 @@ REST_FRAMEWORK__DEFAULT_AUTHENTICATION_CLASSES = (
 ALLOWED_IMPORT_PATHS = <%= scope['pulpcore::allowed_import_path'] %>
 ALLOWED_EXPORT_PATHS = <%= scope['pulpcore::allowed_export_path'] %>
 ALLOWED_CONTENT_CHECKSUMS = <%= scope['pulpcore::allowed_content_checksums'] %>
+<% if scope['pulpcore::use_legacy_tasking_system'] %>
+
+# This setting is necessary but not sufficient to use the new tasking system;
+# the systemd unit files must also be properly configured for either worker type.
+# As such you should NEVER directly modify this setting. Refer to the documentation
+# (from your software vendor, if any, otherwise to the community documentation) for
+# the correct procedure to change between the new and legacy worker types.
```

Comment:
I struggled with writing this comment. I want to issue a warning to users who happen upon the `settings.py` not to modify this setting directly, as updating the setting but not the unit files will result in a broken configuration. But finding the correct way to make the configuration change would be quite different for an upstream user who used the puppet modules directly vs. an upstream user who used foreman-installer vs. an user of a downstream product like Satellite.

Reply:
I suppose warning them is extra helpful. This is a file controlled by Puppet, they should not change *anything* in this file directly. That's a general rule of all things under our control.

Reply:
We should look to add a header to this file if we do not already that it's under Puppet control?

Reply:
With https://github.com/theforeman/puppet-pulpcore/pull/202, is it really needed? The systemd unit will then simply always use the new wrapper that can do both.

Reply:
> I suppose warning them is extra helpful. This is a file controlled by Puppet, they should not change _anything_ in this file directly. That's a general rule of all things under our control.

I am concerned about users falling into the trap of "yes, I know this is managed by the installer, but I don't care about making it persistent, I just want to try out the new workers so I'll change the configuration in settings.py and restart services."

I've seen both end users and support engineers use this type of approach, particularly when troubleshooting (for example to make a temporary change to the # of pulp 2 workers, worker timeout, or download concurrency)

Reply:
> With #202, is it really needed? The systemd unit will then simply always use the new wrapper that can do both.

I think you're right.

I initially misread https://github.com/pulp/pulp_installer/pull/616 / missed https://pulp.plan.io/issues/8721